drgn/helpers/linux/runq.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2025, Oracle and/or its affiliates.
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+"""
+runq
+-------------
+
+The ``drgn.helpers.linux.runq`` module provides helpers for working with the
+Linux runq.
+"""
+from drgn import Object, Program
+from drgn.helpers.linux.percpu import per_cpu
+from drgn.helpers.linux.sched import cpu_curr, task_cpu
+
+
+def runq_clock(prog: Program, cpu: int) -> int:
+    """
+    Get clock of cpu runqueue ``struct rq``
+
+    :param prog: drgn program
+    :param cpu: cpu index
+    :returns: cpu runqueue clock in ns granularity
+    """
+    rq = per_cpu(prog["runqueues"], cpu)
+    return rq.clock.value_()
+
+
+def get_task_arrival_time(task: Object) -> int:
+    """
+    Get a task's arrival time on cpu
+
+    A task's arrival time is only updated when the task is put ON a cpu via
+    context_switch.
+
+    :param task: ``struct task_struct *``
+    :returns: arrival time instance in ns granularity
+    """
+
+    try:
+        arrival_time = task.last_run.value_()
+    except AttributeError:
+        try:
+            arrival_time = task.timestamp.value_()
+        except AttributeError:
+            arrival_time = task.sched_info.last_arrival.value_()
+
+    return arrival_time
```

Comment:
See [this comment](https://github.com/osandov/drgn/pull/519#discussion_r2205958751). I think we don't need this helper. It can be simplified to `task.sched_info.last_arrival` and inlined into the user below.